toolchain/check/handle_impl.cpp

```diff
@@ -374,6 +341,8 @@ static auto BuildImplDecl(Context& context, Parse::AnyImplDeclId node_id,
   // Add the impl declaration.
   bool invalid_redeclaration = false;
   auto lookup_bucket_ref = context.impls().GetOrAddLookupBucket(impl_info);
+  // TODO: detect two impl declarations with the same self type and interface,
```

Comment:
```suggestion
  // TODO: Detect two impl declarations with the same self type and interface,
```

Minor style for [TODO casing](https://google.github.io/styleguide/cppguide.html#TODO_Comments)